### PR TITLE
Cleanup the state package around watchers.

### PR DESCRIPTION
**Pull request by howbazaar**
When the initial split of the watcher type into the txnwatcher and the hubwatcher the old watcher was kept around for the state.Open method. Since then we refactored again to use OpenController, which made sure that there is only ever one low level watcher polling the database for changes.

However state.Open was not cleaned up. This branch removes the state.Open function, and also the watcher type from the state/watcher package.

The DeadWatcher is now a hubwatcher type rather than a watcher. Also there were a bunch of tests that were testing isolation across connections, which really aren't needed, so they have also been cleaned up.

### CODE REVIEW

**Comment by howbazaar** [comment]
charmsSuite.TestCharmsServedSecurely malformed url

!!build!!

**Comment by howbazaar** [comment]
$$merge$$

**Comment by howbazaar** [comment]
$$merge$$

File: state/workers.go [hunk 1]
```diff
@@ -53,20 +59,14 @@ func newWorkers(st *State, hub *pubsub.SimpleHub) (*workers, error) {
 			Clock:        st.clock(),
 		}),
 	}
-	if hub == nil {
-		ws.StartWorker(txnLogWorker, func() (worker.Worker, error) {
-			return watcher.New(st.getTxnLogCollection()), nil
+	ws.StartWorker(txnLogWorker, func() (worker.Worker, error) {
```
Comment (manadart) [inline state/workers.go]: Do we care that none of these `StartWorker` calls are error checked?

Comment (howbazaar) [reply state/workers.go]: A little, because it has been the cause of a few hard to determine bugs. Should get to it sometime.

**Pull request merged**